criu/shmem.c

```diff
@@ -808,20 +809,49 @@ static int dump_one_shmem(struct shmem_info *si)
 {
 	int fd, ret = -1;
 	void *addr;
+	bool seek_data_supported;
 
 	pr_info("Dumping shared memory %ld\n", si->shmid);
 
-	fd = open_proc(si->pid, "map_files/%lx-%lx", si->start, si->end);
-	if (fd < 0)
-		goto err;
+	fd = __open_proc(si->pid, EPERM, O_RDONLY, "map_files/%lx-%lx", si->start, si->end);
+	if (fd >= 0) {
+		addr = mmap(NULL, si->size, PROT_READ, MAP_SHARED, fd, 0);
+		if (addr == MAP_FAILED) {
+			pr_err("Can't map shmem 0x%lx (0x%lx-0x%lx)\n", si->shmid, si->start, si->end);
+			goto errc;
+		}
 
-	addr = mmap(NULL, si->size, PROT_READ, MAP_SHARED, fd, 0);
-	if (addr == MAP_FAILED) {
-		pr_err("Can't map shmem 0x%lx (0x%lx-0x%lx)\n", si->shmid, si->start, si->end);
-		goto errc;
+		seek_data_supported = true;
+	} else {
+		if (errno != EPERM) {
```

Comment:
it looks like the patch description isn't complete. It has to describe of all aspects and reasons.

Reply:
Thanks, I wasn't thinking clearly about what the patch was trying to do. Changed to the following:

```
shmem: Fall back to /proc/$pid/mem if no map_files 

If trying to open /proc/$pid/map_files/x-x for a given VMA fails with
EPERM (can happen in unprivileged mode when running in a non-init user
ns), fall back to reading the content from /proc/$pid/mem.
```

Reply:
Done, it will only fall back to `/proc/$pid/mem` in unprivileged mode and a debug message will be printed:

```
Could not access map_files/ link, falling back to /proc/$pid/mem
```